tests/unittests/test_dev_mode.py

```diff
@@ -0,0 +1,93 @@
+    @mock.patch('tap_deputy.utils.write_config', side_effect=write_config)
+    @mock.patch('tap_deputy.utils.read_config')
+    @mock.patch('tap_deputy.client.DeputyClient.post')
+    def test_dev_mode_not_enabled_expired_token(self, mocked_request, mocked_read_config, mocked_write_config, mocked_json_dump):
+        test_config["expires_at"] = strftime(now() - timedelta(days=1))
+        write_new_config_file()
+        deputy = DeputyClient(config=test_config, config_path=test_config_path, dev_mode=False)
+        old_expiry_date = test_config["expires_at"]
+        mocked_request.return_value = {
+            "refresh_token": "new_refresh_token",
+            "access_token": "new_access_token",
+            "expires_in": 86400
+        }
+        deputy.refresh()
+        
+        self.assertEqual(mocked_read_config.call_count, 1)
+        self.assertEqual(mocked_write_config.call_count, 1)
+        self.assertEqual(deputy.refresh_token, "new_refresh_token")
+        self.assertEqual(deputy.access_token, "new_access_token")
+        self.assertGreater(deputy.expires_at, strptime_to_utc(old_expiry_date))
+        tear_down()
+
+    @mock.patch('tap_deputy.client.DeputyClient.post')
+    def test_dev_mode_not_enabled_valid_token(self, mocked_request):
+        """Verify that in dev mode token validation succeeds with token is not expired"""
+        test_config["expires_at"] = strftime(now() + timedelta(days=1))
+        old_expiry_date = test_config["expires_at"]
+        write_new_config_file()
+        deputy = DeputyClient(config=test_config, config_path=test_config_path, dev_mode=False)
+        self.assertIsNone(deputy.refresh())
+        self.assertEqual(deputy.refresh_token, "old_refresh_token")
+        self.assertEqual(deputy.access_token, "old_access_token")
+        self.assertEqual(deputy.expires_at, strptime_to_utc(old_expiry_date))
+        tear_down()
+
+    def test_dev_mode_enabled_valid_token(self):
+        """Verify that in dev mode token validation succeeds with token is not expired"""
+        test_config["expires_at"] = strftime(now() + timedelta(days=1))
+        write_new_config_file()
+        deputy = DeputyClient(config=test_config, config_path=test_config_path, dev_mode=True)
+        self.assertIsNone(deputy.refresh())
+        self.assertEqual(deputy.refresh_token, "old_refresh_token")
+        self.assertEqual(deputy.access_token, "old_access_token")
+        tear_down()
+
+    def test_dev_mode_enabled_expired_token(self):
+        """Verify that exception is raised when dev mode is enabled and token is expired"""
+        test_config["expires_at"] = strftime(now() - timedelta(days=1))
+        write_new_config_file()
+        deputy = DeputyClient(config=test_config, config_path=test_config_path, dev_mode=True)
+        try:
+            deputy.refresh()
+        except Exception as e:
+            self.assertEqual(str(e), "Access Token in config is expired, unable to authenticate in dev mode")
```

Comment:
Please use [unittest.TestCase.assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)

Reply:
Added the [unittest.TestCase.assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)